time-series/time-series-api/src/main/java/com/powsybl/timeseries/TimeSeriesTable.java

```diff
@@ -640,14 +679,32 @@ private void dumpCache(Writer writer, CsvConfig config, int point, CsvCache cach
         }
     }
 
-    public void writeCsv(Writer writer, char separator, ZoneId zoneId) {
+    private void writeTime(Writer writer, CsvConfig config, int point, int cachedPoint) throws IOException {
+        long time = tableIndex.getTimeAt(point + cachedPoint);
+        switch (config.getTimeFormat()) {
+            case DATE_TIME:
+                ZonedDateTime dateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(time), ZoneId.systemDefault());
+                writer.write(dateTime.format(config.dateTimeFormatter));
+                break;
+            case FRACTIONS_OF_SECOND:
+                Long longTime = new Long(time);
+                double convertedTime = longTime.doubleValue();
+                writer.write(Double.toString(convertedTime / 1000));
+                break;
+            case MILLIS:
+                writer.write(Long.toString(time));
+                break;
```

Comment:
Add a default case that throws an exception

Reply:
Done